tests/foreman/installer/test_installer.py

```diff
@@ -177,20 +181,57 @@ def sat_non_default_install(module_sat_ready_rhels):
         'enable-foreman-plugin-discovery',
         'foreman-proxy-plugin-discovery-install-images true',
     ]
-    install_satellite(module_sat_ready_rhels[1], installer_args, enable_fapolicyd=True)
-    sat = module_sat_ready_rhels[1]
-    sat.enable_ipv6_http_proxy()
-    sat.execute('dnf -y --disableplugin=foreman-protector install foreman-discovery-image')
-    return sat
+
+    def deploy(request=None):
+        rhel_version = get_sat_rhel_version()
+        deploy_args = {
+            'deploy_rhel_version': rhel_version.base_version,
+            'deploy_network_type': 'ipv6' if settings.server.is_ipv6 else 'ipv4',
+            'deploy_flavor': settings.flavors.default,
+            'workflow': settings.server.deploy_workflows.os,
+        }
+        if hasattr(request, 'param'):
+            if isinstance(request.param, dict):
+                deploy_args.update(request.param)
+            else:
+                deploy_args['deploy_rhel_version'] = request.param
+        with Broker(**deploy_args, host_class=Satellite) as sat_host:
+            install_satellite(sat_host, installer_args, enable_fapolicyd=True)
+            sat_host.enable_ipv6_http_proxy()
+            sat_host.execute(
+                'dnf -y --disableplugin=foreman-protector install foreman-discovery-image'
+            )
+            return sat_host
```

Comment:
this this sat_host not being immediately checked in after the return here?

Reply:
Nop the satellite will/should be returned only after all sanity test run is finished

Reply:
what i mean is that when you return, you should be exiting the context manager, which should check back in the satellite. Here's a toy example to illustrate what I mean.

```python
In [1]: class TC:
   ...:     def __enter__(self):
   ...:         print("Entering the context")
   ...:         self.active = True
   ...:         return self
   ...:     def __exit__(self, *tbargs):
   ...:         self.active = False
   ...:         print("Exiting the context")
   ...: 

In [2]: def test_func():
   ...:     with TC() as tc:
   ...:         return tc
   ...: 

In [3]: tc = test_func()
Entering the context
Exiting the context

In [4]: tc.active
Out[4]: False
```

Reply:
@JacobCallahan Do you think the Broker context manager [here](https://github.com/SatelliteQE/robottelo/pull/15948/files#diff-606dfd8ed66a37a0a150b787d8ce417b246c8d3e0aaf8b33c4d35da5189f5792R198) wont checkout the satellite after it finished test execution.

Reply:
@JacobCallahan Earlier it was handled by context manager which was checking out the sat before we operate on it. Now we are explicitly tearing it down in fixture.